universe/Fleet.cpp

```diff
@@ -150,7 +150,8 @@ void Fleet::Copy(std::shared_ptr<const UniverseObject> copied_object, int empire
 
                 const std::list<int>& copied_fleet_route = copied_fleet->m_travel_route;
 
-                if (m_travel_route.empty() && !copied_fleet->m_travel_route.empty())
+                m_travel_route.clear();
+                if (!copied_fleet->m_travel_route.empty())
                     m_travel_route.push_back(moving_to);
```

Comment:
I can't really tell how this is all supposed to work, or be sure that there's not some case this could break, particularly with what's passed into ShortenRouteToEndAtSystem. If you're sure it should work though, I suppose it's OK...